File: 75Review/SlidingWindow/121.go
func maxProfit(prices []int) int {
    l := 0
    profit := 0

    for r := 0; r < len(prices); r++ {
        if prices[r] > prices[l] {
            profit = max(profit, prices[r] - prices[l])
        } else {
            l = r
        }
    }
    return profit
}

File: src/Main/150/121.cc
#include <vector>
using namespace std;

class Solution {
public:
  int maxProfit(vector<int> &prices) {
    int profit = 0;
    int l = 0;
    for (int r = 0; r < prices.size(); r++) {
      if (prices[r] > prices[l]) {
        profit = max(profit, prices[r] - prices[l]);
      } else {
        l = r;
      }
    }
    return profit;
  }
};
